scripts/backport.py

```diff
@@ -0,0 +1,371 @@
+    query = string.Template(
+        """query {
+          repository(owner: "$owner", name: "$name") {
+            pullRequest(number: $pr_number) {
+              id
+            }
+          }
+        }"""
+    ).substitute(pr_number=pr_number, owner=owner, name=name)
+    result = run_query(query)
+    pr_id = result["data"]["repository"]["pullRequest"]["id"]
+
+    query = string.Template(
+        """mutation {
+            enablePullRequestAutoMerge(
+                input: {
+                    pullRequestId: "$pr_id",
+                    mergeMethod: REBASE
+                }
+            ) {
+                clientMutationId
+            }
+        }"""
+    ).substitute(pr_id=pr_id)
+    run_query(query)
+
+
+# The token has to have the "access public repositories" permission, or else creating a PR returns 404.
+github = Github(os.environ.get("GITHUB_TOKEN"))
+
+# If we are running inside Github Action, will modify the main repo.
+source_remote = "origin"
+source_repo_name = os.environ.get("GITHUB_REPOSITORY")
+target_remote = source_remote
+target_repo_name = source_repo_name
+
+if not source_repo_name:
+    # We are running manually for debugging, probably want to modify a fork.
+    source_repo_name = "timescale/timescaledb"
+    target_repo_name = os.environ.get("BACKPORT_TARGET_REPO")
+    target_remote = os.environ.get("BACKPORT_TARGET_REMOTE")
+    if not target_repo_name or not target_remote:
+        print(
+            "Please specify the target repositories for debugging, using the "
+            "environment variables BACKPORT_TARGET_REPO (e.g. `timescale/timescaledb`) "
+            "and BACKPORT_TARGET_REMOTE (e.g. `origin`).",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+print(
+    f"Will look at '{source_repo_name}' (git remote '{source_remote}') for bug fixes, "
+    f"and create the backport PRs in '{target_repo_name}' (git remote '{target_remote}')."
+)
+
+source_repo = github.get_repo(source_repo_name)
+target_repo = github.get_repo(target_repo_name)
+
+# Set git name and email corresponding to the token user.
+token_user = github.get_user()
+os.environ["GIT_COMMITTER_NAME"] = token_user.name
+os.environ[
+    "GIT_COMMITTER_EMAIL"
+] = f"{token_user.id}+{token_user.login}@users.noreply.github.com"
+print(
+    f"Will commit as {os.environ['GIT_COMMITTER_NAME']} <{os.environ['GIT_COMMITTER_EMAIL']}>"
+)
+
+# Fetch the sources
+subprocess.run(f"git fetch {source_remote}", shell=True, check=True)
+subprocess.run(f"git fetch {target_remote}", shell=True, check=True)
+
+# Find out what is the branch corresponding to the previous version compared to
+# main. We will backport to that branch.
+version_config = {
+    x[0].strip(): x[1].strip()
+    for x in [
+        line.split("=")
+        for line in subprocess.check_output(
+            f"git show {source_remote}/main:version.config",
+            shell=True,
+            text=True,
+        ).splitlines()
+        if line
+    ]
+}
```

Comment:
A little easier to read. You can also split it into multiple lines.
```suggestion
# Find out what is the branch corresponding to the previous version compared to
# main. We will backport to that branch.
version_config = dict(
    re.split(r"\s+=\s+", line) 
     for line in subprocess.check_output(
         f"git show {source_remote}/main:version.config",
         shell=True,
         text=True,
     ).splitlines()
     if line
)
```

Reply:
The linter tells me to use dict comprehension instead of `dict()`, but re.split might be nicer indeed, I'll try to incorporate it.

Reply:
No strong opinion, just a suggestion.

Reply:
Wrote it even simpler with capture groups.